presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -1006,7 +1013,16 @@ public long getMaxActiveSplitTime()
     {
         Iterator<RunningSplitInfo> iterator = runningSplitInfos.iterator();
         if (iterator.hasNext()) {
-            return NANOSECONDS.toMillis(ticker.read() - iterator.next().getStartTime());
+            RunningSplitInfo splitInfo = iterator.next();
```

Comment:
Doing the printing inside this method isn't appropriate since it relies on the JMX method being called. Also, this can print the same information repeatedly. Make this a periodic task using `ScheduledExecutorService` and keep track of the last split that was printed, so it's only printed once.